packages/components/src/utils/events.ts

```diff
@@ -0,0 +1,51 @@
+type EventHandler< T extends Event > = ( event: T ) => void;
+
+/**
+ * Merges event handlers together.
+ *
+ * @template TEvent
+ * @param  handler
+ * @param  otherHandler
+ */
+function mergeEvent< TEvent extends Event >(
+	handler: EventHandler< TEvent >,
+	otherHandler: EventHandler< TEvent >
+): EventHandler< TEvent > {
+	return ( event: TEvent ) => {
+		if ( typeof handler === 'function' ) {
+			handler( event );
+		}
+		if ( typeof otherHandler === 'function' ) {
+			otherHandler( event );
+		}
+	};
+}
+
+/**
+ * Merges a set of event handlers together.
+ *
+ * @template TEvent
+ * @param  handlers
+ * @param  extraHandlers
+ */
+export function mergeEventHandlers<
+	TEvent extends Event,
+	TLeft extends Record< string, EventHandler< TEvent > >
+>(
+	handlers: TLeft,
+	extraHandlers: Record< string, EventHandler< TEvent > >
```

Comment:
Why don't `handlers` and `extraHandlers` have the same type?

Reply:
Because of https://github.com/WordPress/gutenberg/pull/33205#issuecomment-875581202

And because they don't actually need to have the same type. I think we should return some kind of mapped type that merges the two.

Reply:
Gotcha. If we go ahead with the changes proposed in https://github.com/WordPress/gutenberg/pull/33205#discussion_r665516453, then I guess we should change these types too, so that we have a `TLeft`, `TRight` and a return value which has all keys of `TLeft` and all keys of `TRight`, correct?

Reply:
Addressed in https://github.com/WordPress/gutenberg/pull/33205/commits/a41f480751de0bb41a9f186266c90435817f706d